l10n_it_ricevute_bancarie/models/account.py

```diff
@@ -38,6 +38,25 @@ class ResPartnerBankAdd(models.Model):
         help="Identification Code of the Company in the Interbank System.",
     )
 
+    def unlink(self):
+        protected_records = (
+            self.env["account.move"]
+            .search([("riba_partner_bank_id", "in", self.ids)])
+            .mapped("riba_partner_bank_id")
+        )
+        records_to_delete = self - protected_records
+
+        if protected_records:
+            message = _(
+                "The bank accounts with accreditation code "
+                f"{[bank.acc_number for bank in protected_records]}"
+                " cannot be deleted as they are used in invoices."
+                " If possible, archive the bank account"
+            )
+            raise UserError(message)
+
+        return super(ResPartnerBankAdd, records_to_delete).unlink()
```

Comment:
```suggestion
    def unlink(self):
        self.check_protected_bank()
        return super().unlink()
        
    def check_protected_bank(self):
          if (
            self.env["account.move"]
            .search([("riba_partner_bank_id", "in", self.ids)])
            .mapped("riba_partner_bank_id")
        ):
            message = _(
                "The bank accounts with accreditation code "
                f"{[bank.acc_number for bank in protected_records]}"
                " cannot be deleted as they are used in invoices."
                " If possible, archive the bank account"
            )
            raise UserError(message)
```


io la vedrei così, ma non so se ho capito esattamente quello che serve